contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -183,6 +205,7 @@ static double timeval_to_double(struct timeval tv)
 
 static int _cb_ruby_wait(trilogy_sock_t *sock, trilogy_wait_t wait)
 {
+    struct trilogy_ctx *ctx = sock->user_data;
```

Comment:
This almost feels like there's a danger of a possible lifetime issue here, but I think it's actually fine.

Reply:
I didn't look too closely at the lifetimes but I assume the socket should outlive the context?